src/d3a/models/state.py

```diff
@@ -234,3 +234,8 @@ def market_cycle(self, past_time_slot, time_slot):
         self.charge_history[time_slot] = 100.0 * self.used_storage / self.capacity
         self.charge_history_kWh[time_slot] = self.used_storage
         self.offered_history[time_slot] = self.offered_sell_kWh[time_slot]
+
+
+class MarketClearingState:
```

Comment:
Can you please explain, why this has to be a separate class? Cant and shouldnt `self.clearing_rate` be a member of TwoSidedPayAsBidEngine and TwoSidedPayAsClearEngine ?

Reply:
I can elaborate a bit more on that, the problem here is that this is not actually a simple variable, but it actually is historical data. The root cause of the issue here was that clearing rate is not actually one value, but multiple ones. In order to denote that this value is for historical data, I find it good to have it in a separate class, and I think it would be good if we did the same thing for the other strategies as well. One step at a time though :)

Reply:
Ok, makes sense! Thanks for the explanation!